vm/src/stdlib/collections.rs

```diff
@@ -321,6 +322,23 @@ mod _collections {
         fn len(&self) -> usize {
             self.borrow_deque().len()
         }
+
+        #[pymethod(magic)]
+        fn iadd(
+            zelf: PyRef<Self>,
+            other: PyObjectRef,
+            vm: &VirtualMachine,
+        ) -> PyResult<PyRef<Self>> {
+            let other: PyObjectRef = if zelf.is(&other) {
+                vm.new_pyobj(zelf.copy())
```

Comment:
by calling here `new_pyobj` and calling `try_from_object` later, this is making unnessessary `PyObjectRef`.